server/datastore/mysql/labels.go

```diff
@@ -165,8 +165,11 @@ VALUES ` + strings.Join(placeholders, ", ")
 		}
 		return nil
 	})
+	if err != nil {
+		return nil, nil, ctxerr.Wrap(ctx, err, "UpdateLabelMembershipByHostIDs transaction")
+	}
 
-	return ctxerr.Wrap(ctx, err, "UpdateLabelMembershipByHostIDs transaction")
+	return ds.labelDB(ctx, labelID, teamFilter, ds.writer(ctx))
```

Comment:
What about `ds.reader(ctx)`, or even using
[this](https://github.com/fleetdm/fleet/blob/b20fe1df83ad8ded36358b4bbb26b60eaaba9d8e/server/datastore/mysql/labels.go#L338) function?

Reply:
The problem with both of those is that they use the read replica, which doesn't have the new label yet.